components/list/list-item-mixin.js

```diff
@@ -0,0 +1,351 @@
+import '../colors/colors.js';
+import './list-item-generic-layout.js';
+import './list-item-placement-marker.js';
+import { css, html } from 'lit-element/lit-element.js';
+import { classMap } from 'lit-html/directives/class-map.js';
+import { getFirstFocusableDescendant } from '../../helpers/focus.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { ListItemCheckboxMixin } from './list-item-checkbox-mixin.js';
+import { ListItemDragDropMixin } from './list-item-drag-drop-mixin.js';
+import { ListItemRoleMixin } from './list-item-role-mixin.js';
+import { nothing } from 'lit-html';
+import ResizeObserver from 'resize-observer-polyfill';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+const ro = new ResizeObserver(entries => {
+	entries.forEach(entry => {
+		if (!entry || !entry.target || !entry.target.resizedCallback) {
+			return;
+		}
+		entry.target.resizedCallback(entry.contentRect && entry.contentRect.width);
+	});
+});
+
+const defaultBreakpoints = [842, 636, 580, 0];
+
+export const ListItemMixin = superclass => class extends ListItemDragDropMixin(ListItemCheckboxMixin(ListItemRoleMixin(RtlMixin(superclass)))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Breakpoints for responsiveness in pixels. There are four different breakpoints and only the four largest breakpoints will be used.
+			 */
+			breakpoints: { type: Array },
+			/**
+			 * Address of item link if navigable
+			 */
+			href: { type: String },
```

Comment:
This will conflict with HM `href`. We will need a new name for this attribute. 

When we do we need to backwards compatible it with `href` on `d2l-list-item` .

The new attribute should also handle javascript methods for click events.

Reply:
For backwards compatibility, we could check if the `href` value is set to a hypermedia entity URL. If it isn't, we can safely assume it is meant as a `d2l-list-item` navigation URL, and fill the new property with that value.

Reply:
I'm thinking, we just update `d2l-list-item` component and handle the backwards compatibility there. Since it will always be a list.